django_cte/query.py

```diff
@@ -75,9 +79,42 @@ def generate_sql(cls, connection, query, as_sql):
         for cte in query._with_ctes:
             if django.VERSION > (4, 2):
                 _ignore_with_col_aliases(cte.query)
-            compiler = cte.query.get_compiler(connection=connection)
+
+            alias = query.alias_map.get(cte.name)
+            is_left_outer = (
+                isinstance(alias, QJoin) and alias.join_type == LOUTER
+            )
+
+            if django.VERSION >= (4, 0):
+                compiler = cte.query.get_compiler(
+                    connection=connection, elide_empty=not is_left_outer
+                )
+            else:
+                compiler = cte.query.get_compiler(connection=connection)
+
             qn = compiler.quote_name_unless_alias
-            cte_sql, cte_params = compiler.as_sql()
+            try:
+                cte_sql, cte_params = compiler.as_sql()
+            except EmptyResultSet:
+                if django.VERSION < (4, 0) and is_left_outer:
+                    # elide_empty is not available prior to Django 4.0. The
+                    # below behavior emulates the logic of it, rebuilding
+                    # the CTE query with a WHERE clause that is always false
+                    # but that the SqlCompiler cannot optimize away. This is
+                    # only required for left outer joins, as standard inner
+                    # joins should be optimized and raise the EmptyResultSet
+                    query = cte.query.copy()
+                    query.where = WhereNode(
+                        [ExtraWhere(["1 = 0"], [])], AND
+                    )
```

Comment:
The `AND` parameter appears to be optional and the default. I didn't check older versions of Django, so maybe this changed within the support window too?
```suggestion
                    query.where = WhereNode([ExtraWhere(["1 = 0"], [])])
```